Bug: release-file bulk deletion can leave files behind

### Describe the bug

The v2 API endpoint for deleting all files for a release can silently leave
files behind when the request includes another valid release-file filter.

`DELETE /api/v2/downloads/release_file/delete_by_release/?release=<id>` is
intended to remove every release file attached to the requested release. The
view currently reuses the normal release-file list filters for the delete
queryset, so an extra valid query parameter such as `os=<id>` narrows the
deletion to only that subset. The endpoint still returns `204`, which makes the
request look successful even though other files for the release remain.

### To Reproduce

1. Create a release with multiple release files, for example files for Windows,
   macOS, and source/Linux.
2. Send an authenticated staff request to:

   ```text
   DELETE /api/v2/downloads/release_file/delete_by_release/?release=<release-id>&os=<one-os-id>
   ```

3. Observe that the response is `204`.
4. Query the release files for that same release:

   ```text
   GET /api/v2/downloads/release_file/?release=<release-id>
   ```

5. Files that did not match the extra `os` filter are still present.

Local regression-test evidence from the existing downloads API fixture:

```text
AssertionError: 3 != 0
```

The request deleted only the Linux/source file and left three other files for
the same release.

### Expected behavior

The delete-by-release action should delete exactly all files attached to the
requested release. Additional valid list filters should not narrow the bulk
delete action. Missing, blank, or malformed `release` values should return
`400 Bad Request` instead of reaching queryset evaluation.

### Additional context

Local regression-test evidence:

```text
FAIL: test_filter_release_file_delete_by_release
AssertionError: 3 != 0
```

The endpoint is used by release automation to replace release-file metadata.
Returning success after a narrowed delete can leave stale metadata rows from the
previous replacement set.

--- a/apps/downloads/api.py
+++ b/apps/downloads/api.py
@@ -177,10 +177,14 @@ class ReleaseFileViewSet(viewsets.ModelViewSet):
     def delete_by_release(self, request):
         """Delete all release files associated with a given release."""
         release = request.query_params.get("release")
-        if release is None:
+        if not release:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        try:
+            release_id = int(release)
+        except ValueError:
             return Response(status=status.HTTP_400_BAD_REQUEST)
         # TODO: We can add support for pagination in the future.
-        queryset = self.filter_queryset(self.get_queryset())
+        queryset = self.get_queryset().filter(release_id=release_id)
         # This calls 'mixins.DestroyModelMixin.perform_destroy()'.
         self.perform_destroy(queryset)
         return Response(status=status.HTTP_204_NO_CONTENT)

--- a/apps/downloads/tests/test_views.py
+++ b/apps/downloads/tests/test_views.py
@@ -547,17 +547,36 @@ class DownloadApiV2ViewsTest(BaseDownloadApiViewsTest, BaseDownloadTests, APITes
             # http://www.django-rest-framework.org/api-guide/viewsets/#reversing-action-urls
             self.create_url(
                 "release_file/delete_by_release",
-                filters={"release": self.release_275.pk},
+                filters={
+                    "release": self.release_275.pk,
+                    "os": self.linux.pk,
+                },
             ),
             HTTP_AUTHORIZATION=self.Authorization,
         )
         self.assertEqual(response.status_code, 204)
 
         # Making a GET request after the deletion shouldn't return any results.
-        response = self.client.get(self.create_url("release_file", filters={"release": self.release_275.pk}))
+        response = self.client.get(
+            self.create_url(
+                "release_file",
+                filters={"release": self.release_275.pk},
+            )
+        )
         self.assertEqual(response.status_code, 200)
         content = self.get_json(response)
         self.assertEqual(len(content), 0)
+
+        # Files for other releases should be left intact.
+        response = self.client.get(
+            self.create_url(
+                "release_file",
+                filters={"release": self.draft_release.pk},
+            )
+        )
+        self.assertEqual(response.status_code, 200)
+        content = self.get_json(response)
+        self.assertEqual(len(content), 1)
 
         # Making a valid request should return 403 Forbidden if it
         # comes from a non-staff user.
@@ -578,6 +597,19 @@ class DownloadApiV2ViewsTest(BaseDownloadApiViewsTest, BaseDownloadTests, APITes
             HTTP_AUTHORIZATION=self.Authorization,
         )
         self.assertEqual(response.status_code, 400)
+
+        # Blank or malformed release values should also return 400 instead of
+        # reaching queryset evaluation with an invalid primary-key value.
+        for invalid_release in ("", "not-a-release-id"):
+            response = self.json_client(
+                "delete",
+                self.create_url(
+                    "release_file/delete_by_release",
+                    filters={"release": invalid_release},
+                ),
+                HTTP_AUTHORIZATION=self.Authorization,
+            )
+            self.assertEqual(response.status_code, 400)
 
         # /release_file/delete_by_release/ should only accept DELETE requests.
         response = self.client.get(

